src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2989,11 +2989,27 @@ internal static bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint
             {
                 case BoundKind.DefaultLiteral:
                 case BoundKind.DefaultExpression:
-                case BoundKind.Parameter:
                 case BoundKind.ThisReference:
                     // always returnable
                     return true;
 
+                case BoundKind.Parameter:
+                    var parameter = ((BoundParameter)expr).ParameterSymbol;
+                    // params Span<T> value cannot be returned since that would
+                    // prevent sharing repeated allocations at the call-site.
+                    if (parameter.IsParams &&
+                        parameter.Ordinal == parameter.ContainingSymbol.GetParameterCount() - 1 &&
+                        !parameter.Type.IsSZArray()) // PROTOTYPE: Should check for ReadOnlySpan<T> or Span<T> explicitly.
+                    {
+                        // PROTOTYPE: Consider reporting the error where the params method is called rather
+                        // than on the definition. That would allow calling this method with an explicit Span<T>
+                        // without an error. That's also important because changing the method defintion between
+                        // params and non-params is not a binary breaking change.
+                        Error(diagnostics, ErrorCode.ERR_EscapeParamsSpan, node, expr.Syntax);
```

Comment:
Added a note to test plan. Per offline conversation, we need to figure out safe escape rules first.
Would it be okay to remove this error in the meantime?

Reply:
Removed.